woocommerce/woocommerce
# [Product Editor] Variation quick actions succeed when sku is inherited from parent

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR addresses the following:

- Variation quick actions (from either the "quick update" menu, or the "..." menu) succeed when the variation inherits its sku from the parent variable product (closes #43800)
- TypeScript updates to require the `id` property in the codebase that handles variation quick actions (since without an `id`, you can't identify what variation to update)
- Always use arrays to pass selection and changes in variation quick actions, to simplify handling (no more `if Array.isArray( selection)` checks everywhere!)



### How to test the changes in this Pull Request:



#### Automated testing

1. Make sure unit tests pass...

```bash
pnpm --filter=@woocommerce/product-editor test:js
```

#### Manual testing

With a WooCommerce dev env with the new product editor enabled (**WooCommerce** > **Settings** > **Advanced** > **Features** > **Experimental features**)...

1. Go to **Products** > **Add New**
2. Add some variations on the **Variations** tab
3. Verify quick actions (both "quick update" and "...") work as expected, properly updating the variations (these should work whether the variation has an explicit sku set or whether it inherits it from the variable parent)

<img width="875" alt="Screenshot 2024-01-23 at 20 55 38" src="https://github.com/woocommerce/woocommerce/assets/2098816/e7050ab0-5ac4-42eb-be49-be7794c5720e">
<img width="875" alt="Screenshot 2024-01-23 at 20 55 27" src="https://github.com/woocommerce/woocommerce/assets/2098816/a4fd1314-7542-448c-802b-e9c92c667ac5">

4. Install the following test plugin, which adds two menu items under the "Pricing" group, and verify that the Slot Fills recently added in #43441 function:

- **Add 100 to regular price (single selection only)**
- **50% off sale (multiple selection allowed)**

[test-44017.zip](https://github.com/woocommerce/woocommerce/files/14054589/test-44017.zip)



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
